lms/djangoapps/courseware/views/index.py

```diff
@@ -177,10 +185,22 @@ def render(self, request):
             self.chapter = self._find_chapter()
             self.section = self._find_section()
 
+            start_date = self.course.start
             if self.chapter and self.section:
                 self._redirect_if_not_requested_section()
                 self._save_positions()
                 self._prefetch_and_bind_section()
+                start_date = self.section.start
+
+            if datetime.now(utc) < start_date and get_course_masquerade(self.effective_user, self.course_key):
+                if is_masquerading_as_student(self.effective_user, self.course_key) or \
+                   (is_masquerading_as_specific_student(self.effective_user, self.course_key) and
+                   not has_staff_roles(self.effective_user, self.course_key)):
```

Comment:
This boolean is a bit of a mouthful, would it be worth splitting it out into a couple of variables?